### PR TITLE
Issue 85 project, dataset and pkpd_model views (incl update)

### PR DESCRIPTION
**Pull request by martinjrobins**
This adds or updates the following views for project, dataset and pkpd_model Models:
- detail
- list
- create
- update
- delete

coverage is still a bit weird, but it says 100% on my machine!

### CODE REVIEW

File: pkpdapp/pkpdapp/models/dataset.py [hunk 1]
```diff
@@ -17,9 +19,10 @@ class Dataset(models.Model):
     ]
     name = models.CharField(max_length=100, help_text='name of the dataset')
     datetime = models.DateTimeField(
+        default=timezone.now,
         help_text=(
-            'Date/time the experiment was conducted. ',
-            'All time measurements are relative to this date/time',
+            'Date/time the experiment was conducted. '
```
Comment (ben18785) [inline pkpdapp/pkpdapp/models/dataset.py:24]: Do we want this as a required field or optional? I would have thought that some of the time this wasn't really relevant if, for example, reusing old data.

File: pkpdapp/pkpdapp/models/dataset.py [hunk 2]
```diff
@@ -17,9 +19,10 @@ class Dataset(models.Model):
...
+            'All time measurements are relative to this date/time'
         )
     )
     description = models.TextField(
```
Comment (ben18785) [inline pkpdapp/pkpdapp/models/dataset.py:28]: I also wonder whether description should be optional?

File: pkpdapp/pkpdapp/templates/dataset_detail.html [hunk 3]
```diff
@@ -6,10 +6,35 @@
...
 
 {% block main_content %}
 <p></p>
 <h2>Dataset {{dataset.name}}</h2>
-<p>{{dataset.description}}</p>
+(<a href="{% url 'dataset-update' dataset.id %}"> Update </a>)
+<p>{{dataset.description|markdownify}}</p>
+<h3>Biomarker Types</h2>
+<ul>
+{% for type in biomarker_types %}
```
Comment (ben18785) [inline pkpdapp/pkpdapp/templates/dataset_detail.html:18]: Do we want to display all the data or just a subset of it? Just think showing (say) 1000 rows might make display a bit better. (I know we'll probably want to do this in a nicer markdowny way later.)

File: pkpdapp/pkpdapp/templates/project_detail.html [hunk 4]
```diff
@@ -6,27 +6,38 @@
...
 
 {% block main_content %}
+
 <p></p>
-<h2>Project {{project.name}}</h2>
-<p>{{project.description}}</p>
+<h2>
+  Project {{project.name}}
+</h2>
+(<a href="{% url 'project-update' project.id %}"> Update </a>)
```
Comment (ben18785) [inline pkpdapp/pkpdapp/templates/project_detail.html:17]: "Update project information"?

Comment (ben18785) [reply pkpdapp/pkpdapp/templates/project_detail.html:17]: I know it's clear to us what it means to update a project but a user doesn't know about the project models...

**Review by ben18785 — APPROVED** [review]
Hi @martinjrobins -- looks good, thanks! Have a look at my comments: these were mostly around whether we wanted some of the fields to be optional. For example, datetime in Dataset -- not sure a user always will want to fill this out (especially if it's old data).

File: pkpdapp/pkpdapp/models/dataset.py (hunk 1 above)
Comment (martinjrobins) [reply pkpdapp/pkpdapp/models/dataset.py:24]: yea, probably should be optional, I'll change that

File: pkpdapp/pkpdapp/models/dataset.py (hunk 2 above)
Comment (martinjrobins) [reply pkpdapp/pkpdapp/models/dataset.py:28]: yup, I'll set that to optional

File: pkpdapp/pkpdapp/templates/dataset_detail.html (hunk 3 above)
Comment (martinjrobins) [reply pkpdapp/pkpdapp/templates/dataset_detail.html:18]: yup, good idea

File: pkpdapp/pkpdapp/templates/project_detail.html (hunk 4 above)
Comment (martinjrobins) [reply pkpdapp/pkpdapp/templates/project_detail.html:17]: yea, that sounds better

File: pkpdapp/pkpdapp/templates/dataset_detail.html (hunk 3 above)
Comment (martinjrobins) [reply pkpdapp/pkpdapp/templates/dataset_detail.html:18]: I've paginated this, so you can click through multiple pages each of length 20

**Review by ben18785 — APPROVED** [review]
Looks great! Thanks @martinjrobins

**Comment by ben18785** [comment]
Would merge but for codecov...not sure what's happening here. When I click through to look at the source of the drop in codecov, I can't seem to find the issue.

**Comment by martinjrobins** [comment]
@DavAug, I think you will have to merge this in cause you have admin rights (might be worth giving admin to me and @ben18785 as well)

**Pull request merged**